### ISSUE TITLE
Provide guidance on delayed payments

### ISSUE THREAD

**Issue by moneyball** [BitcoinDesign/Guide#628]
Sometimes a payment is delayed for potentially hours (and could wind up successful or fail) due to either (a) a mobile wallet that is offline and holds the payment, like Muun or (b) an intermediary routing node for some reason delays payment (likely rare but possible).

The design guide should offer guidance.

**Comment by GBKS** [comment]
So there is some information on the "Sending bitcoin" page under [Transaction processing](https://bitcoin.design/guide/payments/send/#transaction-processing) and [Errors](https://bitcoin.design/guide/payments/send/#errors). I suspect this may not be as much detail as you are looking for, but please take a look and let me know what's missing.

**Comment by Bosch-0** [comment]
Is it possible to cancel a Lightning payment and try a different route (if that is the issue). Could be an option if payment gets stuck.

**Comment by GBKS** [comment]
@Bosch-0 according to [this StackExchange reply](https://bitcoin.stackexchange.com/a/86229/109373) from 2019, cancelling a transaction en route is not possible.

In addition to the two spots I mentioned in my previous comment, there's also a reference to delayed payments in the [status display paragraph](https://bitcoin.design/guide/daily-spending-wallet/activity/#status-display) of the Activity page. A little bit spread out, but I think that should cover it. Here the three spots again:
- [Transaction processing](https://bitcoin.design/guide/daily-spending-wallet/sending/#transaction-processing) talks about showing the delayed status on the invoice screen
- [Errors](https://bitcoin.design/guide/daily-spending-wallet/sending/#errors) shows a mockup for an error screen when a payment fails (including options to retry and contact the wallet owner)
- [Status display](https://bitcoin.design/guide/daily-spending-wallet/activity/#status-display) summarizes various states a transaction can be in

Does that satisfy the request?

**Comment by GBKS** [comment]
Interesting, had not seen those before. If I understand correctly, it should be possible to see this "accepted but not settled" state and show it in the UI.

**Comment by Bosch-0** [comment]
Yeah it would be something like that on the UI level. @johnsBeharry has a bit of a deeper dive on them [here](https://bolt.fun/guide/invoices/hodl-invoice) too. 

There would also be a "canceled" state if the invoice expires before the recipient releases the preimage to the sender.

**Comment by GBKS** [comment]
Can the application know whether it was cancelled or expired? Meaning whether this can be distinguished on a  technical level.

**Comment by sbddesign** [comment]
> Can the application know whether it was cancelled or expired? Meaning whether this can be distinguished on a technical level.

@GBKS Short answer: Yes.

Long Answer: if you want a deeper understanding, think back to the HTLCs we learned about while reading Mastering Lightning together. If the recipient fails to provide a preimage, then all the HTLCs between all the prior hops "unwind", and so the payment has failed atomically.

With a HOLD / HODL invoice, the lock-time on the HTLC for the final hop is significantly larger than that of the other HTLCs in the path, giving the recipient time to hold on and decide if they want to accept the payment or not. If they decide they want to accept it during this time window, they reveal the preimage. If they don't reveal the preimage, then the payment fails atomically.

It's really the exact same mechanism by which "regular" lightning payments fail -- the only difference is that the final hop has a longer expiry on their HTLC. They say that the sender has no way of even knowing for sure if it's a hodl invoice.

My guess is that this is achieved using the `min_final_cltv_expiry` in a BOLT11 invoice, but maybe someone can fact check that for me.

**Comment by GBKS** [comment]
@sbddesign sounds like it needs to be inferred whether it's a HOLD invoice by assuming it is uniquely defined by a long expiry on the last hop?

**Comment by sbddesign** [comment]
When you say "inferred", I'm assuming you are referring to the recipient. I think that from the recipient's perspective, this may just look like a "stuck HTLC". In other words, I've routed the payment, and I have not received a preimage or a message telling me that the payment failed. Therefore, I have to treat it as an in-flight HTLC.

Maybe this stuck HTLC is because a node went down on the payment path without responding with a failure message. Maybe it's because this was a HODL invoice.

I think most wallets just show this as "Pending" in their Activity list.

Is there a way to differentiate between an invoice being "held" and any other stuck HTLC? Not sure.

**Comment by GBKS** [comment]
That's exactly what I was referring to. We have a "pending" state already, so maybe that's good enough?

I think it's also up to the requester to state how payment acceptance will be handled. For example, the check-out page of an online store could put an appropriate message below the payment info.

**Comment by sbddesign** [comment]
> We have a "pending" state already, so maybe that's good enough?

I think that's probably good enough.

Just to use our imagination for a moment: it would arguably be a better UX if it could say something like "Bob's Widget Corp. has received your payment, but has not chosen to redeem if yet." Even cooler if it could also say "For more information, see your order details at `https://bobswidget ... /my-orders/uuid...`."

But since a hodl invoice is "[just a regular invoice](https://docs.lightning.engineering/community-resources/glossary#hodl-invoice)", I really don't see the ability for adding such rich metadata at present.

**Comment by Bosch-0** [comment]
Something else we _may_ need to cover is receiving offline payments from an LSP or LWS (using something like greenlight). They are essentially pending incoming payments

**Comment by sbddesign** [comment]
I don't follow. If I'm offline, that means I'm probably not using the app. When I open the app, I become online, and so I receive the payment from the LSP or Greenlight.

Not sure at what point the payment would show up "pending" in that scenario. Am I missing something?

**Comment by Bosch-0** [comment]
You could get a notification from the LSP, greenlight or whatever you are using that you have a payment waiting for you to log in and receive

**Comment by GBKS** [comment]
I'd like to close this issue out. The notification when receiving could go into the Requesting/Receive page you are working on, @Bosch-0 .

**Comment by Bosch-0** [comment]
Yep will include this in the receive page. Would also want something included on the sending side about this.

**Comment by GBKS** [comment]
Great. Closing this now out now. If there are other nuances of delayed payments that we should cover, please open a new issue with the specific request.
